sync/rpc/server.go

```diff
@@ -308,3 +328,91 @@ func (*SyncAgentServer) BackupRestoreIncrementally(ctx context.Context, req *Bac
 	logrus.Infof("Done running %s %v", "sbackup", cmd.Args)
 	return &Empty{}, nil
 }
+
+//The following APIs implement the CRUD interface for the backup object
+// The slice backuplist is implemented similar to a FIFO queue.
+// Insertion of new BackupInfo object happens in the rear end of backuplist
+// BackupUpdate deletes all the old completed backups from the front
+// BackupDelete will delete the entry with the corresponding backupID
+
+func (b *Backup) BackupCreate(backupID string, backup *replica.Backup) {
+	b.Lock()
+	defer b.Unlock()
+
+	b.backupList = append(b.backupList, &BackupInfo{
+		backupID:     backupID,
+		backupStatus: backup,
+	})
+}
+
+func (b *Backup) BackupGet(backupID string) (*replica.Backup, error) {
+	if backupID == "" {
+		return nil, fmt.Errorf("empty backupID")
+	}
+	b.Lock()
+	defer b.Unlock()
+
+	for _, info := range b.backupList {
+		if info.backupID == backupID {
+			return info.backupStatus, nil
+		}
+	}
+	return nil, fmt.Errorf("backup not found %v", backupID)
+}
+
+func (*Backup) remove(b []*BackupInfo, index int) ([]*BackupInfo, error) {
+	if index >= len(b) || index < 0 {
+		return nil, fmt.Errorf("index out of range")
+	}
+	return append(b[:index], b[index+1:]...), nil
+}
+
+func (b *Backup) BackupUpdate() error {
```

Comment:
As I mentioned before, no need to return an error, and there is no error return by the function.

Reply:
Error is being returned inside the function at line 392. 
Should we account for the fact that if there are any issue encountered, say indices go out of range when I call to delete from slice ?
From manual testing, I didn't see any such issues, but it's necessary to have them in the code right?